src/passes/Outlining.cpp

```diff
@@ -261,8 +261,17 @@ struct Outlining : public Pass {
          exprIdx++) {
       sig += StackSignature(exprs[exprIdx]);
     }
-    module->addFunction(Builder::makeFunction(
-      outlinedFunc, Signature(sig.params, sig.results), {}));
+    // Purposefully inserting the outlined functions at the beginning of the
+    // functions vector, so that outlined function assertions are positioned
+    // within the test module. This greatly improves readability of the
+    // outlining lit tests. Because of the direct manipulation of the functions
+    // vector, instead of using the addFunction() helper method, a call to
+    // updateFunctionsMap() is required.
+    module->functions.insert(
+      module->functions.begin(),
+      Builder::makeFunction(
+        outlinedFunc, Signature(sig.params, sig.results), {}));
+    module->updateFunctionsMap();
```

Comment:
It would be better to update the function map just once at the end rather than for each outlined function. How about we rename this function `createOutlinedFunction` and have it return the `std::unique_ptr<Function>` it gets from `Builder::makeFunction`. Then in the caller we could insert all the new functions into `module->functions` and update the functions map in one go.

Reply:
great idea, thanks